weblate/api/tests.py

```diff
@@ -1235,6 +1237,54 @@ def test_patch(self):
         )
         self.assertEqual(response.data["slug"], "new-slug")
 
+    def test_create_component_docfile(self):
+        with open(TEST_DOC, "rb") as handle:
+            response = self.do_request(
+                "api:project-components",
+                self.project_kwargs,
+                method="post",
+                code=201,
+                superuser=True,
+                request={
+                    "docfile": handle,
+                    "name": "Local project",
+                    "slug": "local-project",
+                    "repo": "local:",
+                    "vcs": "local",
+                    "filemask": "*.strings",
+                    "template": "en.strings",
+                    "file_format": "strings-utf8",
+                    "push": "https://username:password@github.com/example/push.git",
+                    "new_lang": "none",
```

Comment:
As there are no translations existing when uploading a single file, adding of translations should be supported.

```suggestion
                    "new_lang": "add",
```